pkg/sysregistriesv2/system_registries_v2.go

```diff
@@ -71,6 +75,42 @@ type Registry struct {
 	Prefix string `toml:"prefix"`
 }
 
+// PullSource consists of an Endpoint and a Reference. Note that the reference is
+// rewritten according to the registries prefix and the Endpoint's location.
+type PullSource struct {
+	Endpoint  Endpoint
+	Reference reference.Named
+}
+
+// PullSourcesFromReference returns a slice of PullSource's based on the passed
+// reference.
+func (r *Registry) PullSourcesFromReference(ref reference.Named) ([]PullSource, error) {
+	if r.MirrorByDigestOnly {
+		// Return the endpoint if we only allow mirror digest pulls and the
+		// passed reference includes a digest.
+		if _, isDigested := ref.(reference.Canonical); isDigested {
```

Comment:
Shouldn’t this be `!isDigested`? The tests also show that the logic is reversed.

Reply:
Complete brain fart on my side. Thanks for catching that!